lld/COFF/Symbols.h

```diff
@@ -340,7 +340,10 @@ class Undefined : public Symbol {
   // If this symbol is external weak, try to resolve it to a defined
   // symbol by searching the chain of fallback symbols. Returns the symbol if
   // successful, otherwise returns null.
-  Defined *getWeakAlias();
+  Symbol *getWeakAlias();
```

Comment:
In trying to read the diff and the effects of it - I see that a number of callers are switched from `getWeakAlias()` to `getDefinedWeakAlias()` - but what I don't see from the point of view of the diff, is what remaining callers there are, that still are calling `getWeakAlias()`?

Because any caller that is switched to `getDefinedWeakAlias()` will essentially be doing the same as before (minus the UB in `cast<Undefined>()`), but which callers do get the new behaviour of `getWeakAlias()` where it potentially returns something else?

Reply:
The remaining callers use `getWeakAlias` to check if a symbol can be resolved. After the `run()` call, all such symbols should be replaced with a `Defined` symbol (assuming valid input, though a malformed archive index could disrupt this). This means that for many callers, `getWeakAlias` isn’t expected to handle lazy symbols, making it functionally equivalent to `getDefinedWeakAlias`. They only differ if new weak aliases are defined between `run()` calls.

In the test example from this PR, using `getDefinedWeakAlias` in `findMangle` would mostly work as well, but it's a bit more fragile. Processing `-export:func` would create a weak alias to `_func@0`, but `-export:func2=func` would still consider it unresolvable and attempt to mangle it again. On the other hand, `getWeakAlias` returns the lazy symbol, so we know it will be resolvable and can skip the unnecessary mangling attempt.